Bitmex.NET/BitmexApiResult.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+
+namespace Bitmex.NET
+{
+    public abstract class BitmexApiResultBase
+    {
+        /// <summary>
+        /// Current rate limit
+        /// </summary>
+        public int RateLimitLimit { get; set; }
+
+        /// <summary>
+        /// Remaining requests
+        /// </summary>
+        public int RateLimitRemaining { get; set; }
+
+        /// <summary>
+        /// Time at which you will have enough requests left to retry your current request
+        /// </summary>
+        public DateTime RateLimitReset { get; set; }
+    }
+
+    public class BitmexApiResult<TResult> : BitmexApiResultBase
+    {
+        /// <summary>
+        /// The result
+        /// </summary>
+        public TResult Result { get; set; }
```

Comment:
@HA07Z7XE it would be great to make  `Result ` and all other properties of the base class immutable.  having a public setter for them makes no sense and can entangle users of our lib

Reply:
done